api/envoy/extensions/quic/server_preferred_address/v3/fixed_server_preferred_address_config.proto

```diff
@@ -21,15 +23,43 @@ message FixedServerPreferredAddressConfig {
 
   option (xds.annotations.v3.message_status).work_in_progress = true;
 
-  oneof ipv4_type {
-    // String representation of IPv4 address, i.e. "127.0.0.2".
-    // If not specified, none will be configured.
-    string ipv4_address = 1;
-  }
+  // Addresses for server preferred address for a single address family (IPv4 or IPv6).
+  message AddressFamilyConfig {
+    // The server preferred address sent to clients.
+    //
+    // Note: Envoy currently must receive all packets for a QUIC connection on the same port, so unless
+    // :ref:`dnat_address <envoy_v3_api_field_extensions.quic.server_preferred_address.v3.FixedServerPreferredAddressConfig.AddressFamilyConfig.dnat_address>`
+    // is configured, the port for this address must be zero, and the listener's
+    // port will be used instead.
+    config.core.v3.SocketAddress address = 1;
 
-  oneof ipv6_type {
-    // String representation of IPv6 address, i.e. "::1".
-    // If not specified, none will be configured.
-    string ipv6_address = 2;
+    // If there is a DNAT between the client and Envoy, the address that Envoy will observe
+    // server preferred address packets being sent to. If this is not specified, it is assumed
+    // there is no DNAT and the server preferred address packets will be sent to the address advertised
+    // to clients for server preferred address.
+    //
+    // Note: Envoy currently must receive all packets for a QUIC connection on the same port, so the
+    // port for this address must be zero, and the listener's port will be used instead.
+    config.core.v3.SocketAddress dnat_address = 2;
   }
+
+  // String representation of IPv4 address, i.e. "127.0.0.2".
+  // The listener's port will be used.
```

Comment:
We don't support a different port today, so maybe not explicitly mention it here, or explicitly say that the current implementation will not with a different port.